game_env.py

```diff
@@ -897,6 +897,73 @@ def play(self):
+        """
+        Returns a dictionary containing transition list of each transformation of the board
+        Total of 12 representation are possible 4 sides of the board X 3 views - normal, horizontal-flip
+        vertical - flip
+        # takes 3ms per transition list
+
+        Returns
+        --------
+        transition_dict - Dictionary consisting of transition list of each representation of the board  
+
+        """
+
+        s, legal_moves, current_player, a, next_s,\
+        next_legal_moves, next_player, done, winner = transition_list
+        
+        s1_nd = self._converter.convert(s[0], input_format='bitboard_single', output_format='ndarray')
+        s2_nd = self._converter.convert(s[1], input_format='bitboard_single', output_format='ndarray')
+        legal_moves_nd = self._converter.convert(legal_moves, input_format='bitboard_single',\
+                                                 output_format='ndarray')
+        a_nd = self._converter.convert(a, input_format='bitboard_single', output_format='ndarray')
+        next_s1_nd = self._converter.convert(next_s[0], input_format='bitboard_single',\
+                                             output_format='ndarray')
+        next_s2_nd = self._converter.convert(next_s[1], input_format='bitboard_single',\
+                                             output_format='ndarray')
+        next_legal_moves_nd = self._converter.convert(next_legal_moves, input_format='bitboard_single',\
+                                                      output_format='ndarray')
+        nd_list = [s1_nd, s2_nd, legal_moves_nd, a_nd, next_s1_nd, next_s2_nd, next_legal_moves_nd]
+
+        transition_dict = {}
+        transition_dict['normal_bottom'] = transition_list
+
+        self._nd_trans_fn = {'normal_left': lambda x: np.rot90(x, 1),
+                             'normal_top': lambda x: np.rot90(x, 2),
+                             'normal_right': lambda x: np.rot90(x, 3),
+                             'horizontal_bottom': lambda x: np.flip(x, axis=0),
+                             'horizontal_left': lambda x: np.rot90(np.flip(x, axis=0), 1),
+                             'horizontal_top': lambda x: np.rot90(np.flip(x, axis=0), 2),
+                             'horizontal_right': lambda x: np.rot90(np.flip(x, axis=0), 3),
+                             'vertical_bottom': lambda x: np.flip(x, axis=1),
+                             'vertical_left': lambda x: np.rot90(np.flip(x, axis=1), 1),
+                             'vertical_top': lambda x: np.rot90(np.flip(x, axis=1), 2),
+                             'vertical_right': lambda x: np.rot90(np.flip(x, axis=1), 3)}
+
+        for key in self._nd_trans_fn.keys():
+            temp_nd_list = [self._nd_trans_fn[key](nd_arr) for nd_arr in nd_list]
+            transition_dict[key] = [
+            # transformed states list
+            [int(np.multiply(temp_nd_list[0], self._converter._array_to_bitboard).sum()),
```

Comment:
better to use the converter functions rather than a private object of the converter class, this way even if some modifications/optimizations are made later, the function can still be used as is without modifying any of the code using the function